x/evm/keeper/static_precompiles.go

```diff
@@ -0,0 +1,228 @@
+	erc20precompile "github.com/evmos/evmos/v16/precompiles/erc20"
+	ics20precompile "github.com/evmos/evmos/v16/precompiles/ics20"
+	osmosisoutpost "github.com/evmos/evmos/v16/precompiles/outposts/osmosis"
+	strideoutpost "github.com/evmos/evmos/v16/precompiles/outposts/stride"
+	"github.com/evmos/evmos/v16/precompiles/p256"
+	stakingprecompile "github.com/evmos/evmos/v16/precompiles/staking"
+	vestingprecompile "github.com/evmos/evmos/v16/precompiles/vesting"
+	"github.com/evmos/evmos/v16/precompiles/werc20"
+	"github.com/evmos/evmos/v16/utils"
+	erc20Keeper "github.com/evmos/evmos/v16/x/erc20/keeper"
+	erc20types "github.com/evmos/evmos/v16/x/erc20/types"
+	"github.com/evmos/evmos/v16/x/evm/types"
+	transferkeeper "github.com/evmos/evmos/v16/x/ibc/transfer/keeper"
+	vestingkeeper "github.com/evmos/evmos/v16/x/vesting/keeper"
+	"golang.org/x/exp/maps"
+)
+
+// AvailableStaticPrecompiles returns the list of all available precompiled contracts.
+// NOTE: this should only be used during initialization of the Keeper.
+func AvailableStaticPrecompiles(
+	chainID string,
+	stakingKeeper stakingkeeper.Keeper,
+	distributionKeeper distributionkeeper.Keeper,
+	bankKeeper bankkeeper.Keeper,
+	erc20Keeper erc20Keeper.Keeper,
+	vestingKeeper vestingkeeper.Keeper,
+	authzKeeper authzkeeper.Keeper,
+	transferKeeper transferkeeper.Keeper,
+	channelKeeper channelkeeper.Keeper,
+) map[common.Address]vm.PrecompiledContract {
+	// Clone the mapping from the latest EVM fork.
+	precompiles := maps.Clone(vm.PrecompiledContractsBerlin)
+
+	// secp256r1 precompile as per EIP-7212
+	p256Precompile := &p256.Precompile{}
+
+	bech32Precompile, err := bech32.NewPrecompile(6000)
+	if err != nil {
+		panic(fmt.Errorf("failed to instantiate bech32 precompile: %w", err))
+	}
+
+	stakingPrecompile, err := stakingprecompile.NewPrecompile(stakingKeeper, authzKeeper)
+	if err != nil {
+		panic(fmt.Errorf("failed to instantiate staking precompile: %w", err))
+	}
+
+	distributionPrecompile, err := distprecompile.NewPrecompile(distributionKeeper, stakingKeeper, authzKeeper)
+	if err != nil {
+		panic(fmt.Errorf("failed to instantiate distribution precompile: %w", err))
+	}
+
+	ibcTransferPrecompile, err := ics20precompile.NewPrecompile(transferKeeper, channelKeeper, authzKeeper)
+	if err != nil {
+		panic(fmt.Errorf("failed to instantiate ICS20 precompile: %w", err))
+	}
+
+	vestingPrecompile, err := vestingprecompile.NewPrecompile(vestingKeeper, authzKeeper)
+	if err != nil {
+		panic(fmt.Errorf("failed to instantiate vesting precompile: %w", err))
+	}
+
+	bankPrecompile, err := bankprecompile.NewPrecompile(bankKeeper, erc20Keeper)
+	if err != nil {
+		panic(fmt.Errorf("failed to instantiate bank precompile: %w", err))
+	}
+
+	var WEVMOSAddress common.Address
+	if utils.IsMainnet(chainID) {
+		WEVMOSAddress = common.HexToAddress(erc20precompile.WEVMOSContractMainnet)
+	} else {
+		WEVMOSAddress = common.HexToAddress(erc20precompile.WEVMOSContractTestnet)
+	}
```

Comment:
There is no safe guard currently for a wrong chain ID - do we want to return an error if that's the case?

Reply:
This wont be used anymore as it was part of the Outposts where we were hardcoding WEVMOS address. Will be removed in the cleanup PR that's currently in draft https://github.com/evmos/evmos/pull/2323